pandas/core/internals.py

```diff
@@ -2914,35 +2914,63 @@ def sparse_reindex(self, new_index):
                                           placement=self.mgr_locs)
 
 
+def get_block_type(values, dtype=None):
+    """
+    Find the appropriate Block subclass to use for the given values and dtype.
+
+    Parameters
+    ----------
+    values : ndarray-like
+    dtype : numpy or pandas dtype
+
+    Returns
+    -------
+    block_type : str
+    """
+    dtype = dtype or values.dtype
+    vtype = dtype.type
+
+    if is_sparse(values):
+        block_type = 'sparse'
+    elif issubclass(vtype, np.floating):
+        block_type = 'float'
```

Comment:
this seems even more complicated. just return the block type klass directly here, no need to return a string which we have a dict for.